src/backend/joanie/signature/backends/dummy.py

```diff
@@ -39,21 +39,9 @@ def submit_for_signature(self, title: str, file_bytes: bytes, order: models.Orde
 
     def get_signature_invitation_link(self, recipient_email: str, reference_ids: list):
         """
-        Dummy method that prepares an invitation link, and it triggers an email notifying that the
-        file is available to download to the signer by email.
+        Dummy method that prepares an invitation link.
```

Comment:
Who calls now the method `_send_mail()` ?  Else, it seems like we are not using the `_send_mail()` method anymore.
IMO, we should trigger the email once the organization has signed the contract, to mock the real signature provider behaviour. Since it's for dummy, I would trigger the email before this [line](https://github.com/openfun/joanie/blob/83c9dbff847f17158eaf7e37afd332b74f2b22a2/src/backend/joanie/signature/backends/dummy.py#L68)

I know it's tricky, because in the updated test, we directly call the method `confirm_student_signature()`...

Reply:
I don't think we should send any mail in dummy if none is sent by a real backend. WDYT ?

Reply:
I think it's appropriate for our context of the dummy class.
In real life, it only sends the email with the link to download the file once the document is fully signed.
Let's make sure to take away in the existing tests docstrings that it sends an email then.